### ISSUE TITLE
Issue with WITH DISTINCT on 3.5.0 throwing 'key not found' error for existing keys

### ISSUE THREAD

**Issue by Clooney24** [neo4j/neo4j#12105]
We are facing a strange behaviour in our 3.5.0 test database. The following cypher returns as expected one existing Uuid:

```
MATCH (industry:IndustryCM)
WHERE industry.Uuid = "72eb68e8-2418-4039-896e-799f497bb7ab" 
WITH industry
RETURN industry.Uuid
```

Almost the same cypher just including DISTINCT in the WITH clause:
```
MATCH (industry:IndustryCM)
WHERE industry.Uuid = "72eb68e8-2418-4039-896e-799f497bb7ab" 
WITH DISTINCT industry
RETURN industry.Uuid
```

returns the following error:

> Neo.DatabaseError.General.UnknownError: key not found: CachedNodeProperty(industry,PropertyKeyName(Uuid))

This happens on our test db (3.5.0) running a backup from the live db (3.4.7) - so same data.

Executing the same cypher on our live db (3.4.7) on the same data works fine.

Just tested using DISTINCT in combination with RETURN instead of WITH -  this works fine on 3.5.0:

```
MATCH (industry:IndustryCM)
WHERE industry.Uuid = "72eb68e8-2418-4039-896e-799f497bb7ab" 
RETURN DISTINCT industry.Uuid
```

We need the WITH DISTINCT statement as the cypher in our scenario is a few lines longer.

I appreciate your help.

**Comment by Hunterness** [comment]
Hi, thanks for reporting. We will investigate and fix this.

**Comment by Lojjs** [comment]
@Clooney24 This is an unfortunate bug that sneaked into 3.5.0. A fix has just been merged and will be included in 3.5.1.

Best regards Louise, Neo4j Cypher team
